### ISSUE TITLE
Arch XFCE4 Pulseeffects-5.0.0-1, with PipeWire, how do I control volume and prevent feedback loops?

### ISSUE THREAD

**Issue by seedship** [wwmm/pulseeffects#892]
Before the update to PulseWire, I used to set built in audio to 100%, and then change the gain on PulseEffects sink. Because when I set my audio device to PulseEffects, it would use the Pulseeffects gain.
![image](https://user-images.githubusercontent.com/885667/106465307-154c5f80-649a-11eb-9b92-6f51792bffaa.png)
Now, changing the gain on PulseEffects sink does nothing, and I need to change the volume of built in audio. However, if I change the audio source to built in audio, it bypasses pulseeffects.
![image](https://user-images.githubusercontent.com/885667/106465566-6ceacb00-649a-11eb-8426-1541e2e53113.png)

In addition, when I switch from built in audio to pulseeffects, it creates this feedback loop where the output of pulseeffects goes right back into the input of pulseeffects instead of my system speakers, and I have to change the output to my speakers.
![image](https://user-images.githubusercontent.com/885667/106465966-00240080-649b-11eb-8d28-de0ccf0bd13c.png)

Do you have any advice on how to properly set up the volume and outputs so I can use my audio buttons on my keyboard, instead of opening up my advanced volume settings and changing all these settings? Thanks!

I am currently running pulseeffects application separately, and not starting the daemon it on startup.

**Comment by seedship** [comment]
Ah, I did not have `pipewire-pulse`. However, now I have it, but I don't know how to control the volume. Before the update to PulseWire, I used to set built in audio to 100%, and then change the gain on PulseEffects sink. Now, changing the gain on PulseEffects sink does nothing, and I need to change the volume of built in audio. However, if I change the audio source to built in audio, it bypasses pulseeffects.
![image](https://user-images.githubusercontent.com/885667/106465307-154c5f80-649a-11eb-9b92-6f51792bffaa.png)
![image](https://user-images.githubusercontent.com/885667/106465566-6ceacb00-649a-11eb-8426-1541e2e53113.png)

Do you have any advice on how to properly set up the volume so I can use my audio buttons on my keyboard, instead of opening up my advanced volume settings and changing the gains on each device? Thanks!

**Comment by wwmm** [comment]
>changing the gain on PulseEffects sink does nothing

Same on my computer. This is probably a bug in PipeWire's compatibility layer for Pulseaudio applications.

>However, if I change the audio source to built in audio, it bypasses pulseeffects.

That should not happen. The volume keys on my keyboard control the volume of my sound card just like it did on Pulseaudio. Based on your Pavucontrol image you are setting `PulseEffects Sink` as default device. Usually it is not a big deal but I do not recommend that. It is better to keep your sound card as default device and enabling `Process All Outputs` in PulseEffects settings menu. What is the default.

**Comment by wwmm** [comment]
>However, if I change the audio source to built in audio, it bypasses pulseeffects.

Your applet is probably forcing all streams to be moved to the new default device instead of just setting the default. Restarting PulseEffects after that or the audio applications should make sound go through PulseEffects again. OR you can use Pavucontrol to set the default instead.

**Comment by wwmm** [comment]
>Do you have any advice on how to properly set up the volume and outputs so I can use my audio buttons on my keyboard

Not setting `PulseEffects Sink` as default should be enough to have your volume keys controlling the sound card again.

**Comment by wwmm** [comment]
>it creates this feedback loop 

PulseEffects stream in Pavucontrol Playback tab should be playing to your sound card. This unusual setting was probably forced by your device managre applet. That is why I say it is bad that software like this one and Gnome control center move the stream instead of just setting the default device. They break custom audio routing in the process =/

**Comment by wwmm** [comment]
It is not related to this issue but if you moved to PipeWire only because we did it be aware that there is a Pulseaudio branch still being maintained https://github.com/wwmm/pulseeffects/tree/pulseaudio-legacy. I think that there are pkgbuild on the aur for it too besides the one in our repository.

**Comment by seedship** [comment]
> > it creates this feedback loop
> 
> PulseEffects stream in Pavucontrol Playback tab should be playing to your sound card. This unusual setting was probably forced by your device manager applet. That is why I say it is bad that software like this one and Gnome control center move the stream instead of just setting the default device. They break custom audio routing in the process =/

Ahh makes sense. I always found my device manager annoying because of that but I got used to it after a while haha. Thanks for the information! Yeah, I found that enabling pulseeffects at startup worked, but if I ever wanted to change to default built in audio and then back to pulse effects, I had to go through all that annoyingness again to get the settings back. Yeah, I did move to PipeWire just because PulseEffects did, but I think for now I'll go back and stick to the legacy with PulseEffects.

Thanks again for developing this and being so responsive! I really love this project!
